src/gui/screen_menu_esp_update.cpp

```diff
@@ -84,6 +85,9 @@ class ScreenMenuESPUpdate : public AddSuperWindow<screen_t> {
     esp_entry *current_file;
     uint32_t readCount;
     loader_stm32_config_t loader_config;
+    std::unique_ptr<char[]> progress_buffer;
```

Comment:
use `std::array` instead , there is filename length limit somewhere in fatfs
there is no dynamic allocation in GUI thread and I would like to stay that way
dynamic allocation is causing RAM fragmentation and you must be able to handle worse case scenario anyway